app/src/org/commcare/android/view/SquareButton.java

```diff
@@ -27,7 +27,12 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         int w = MeasureSpec.getSize(widthMeasureSpec);
         int h = MeasureSpec.getSize(heightMeasureSpec);
 
-        w = Math.max(w, h);
+        if(w == 0 || h == 0) { // this can happen in the home screen gridView, and then we'll have an invisible view
```

Comment:
This comment is pretty confusing. When do we have an invisible view? Are we avoiding an invisible view?